src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts

```diff
@@ -87,41 +88,62 @@ export class NotificationService {
    * @param {string} [message] The message to be displayed. Note, use this field
    *   for error notifications only.
    * @param {*} [options] toastr compatible options, used when creating a toastr
+   * @param {string} [application] Only needed if notification comes from an external application
    * @returns The timeout ID that is set to be able to cancel the notification.
    */
-  show(type: NotificationType, title: string, message?: string, options?: any): number;
-  show(config: CdNotificationConfig): number;
   show(
-    arg: NotificationType | CdNotificationConfig,
+    type: NotificationType,
+    title: string,
+    message?: string,
+    options?: any | ToastOptions,
+    application?: string
+  ): number;
+  show(config: CdNotificationConfig | (() => CdNotificationConfig)): number;
+  show(
+    arg: NotificationType | CdNotificationConfig | (() => CdNotificationConfig),
     title?: string,
     message?: string,
-    options?: any
+    options?: any | ToastOptions,
+    application?: string
   ): number {
-    let type;
-    if (_.isObject(arg)) {
-      ({ message, type, title, options } = <CdNotificationConfig>arg);
-    } else {
-      type = arg;
-    }
     return window.setTimeout(() => {
-      this.save(type, title, message);
-      if (!message) {
-        message = '';
-      }
-      switch (type) {
-        case NotificationType.error:
-          this.toastr.error(message, title, options);
-          break;
-        case NotificationType.info:
-          this.toastr.info(message, title, options);
-          break;
-        case NotificationType.success:
-          this.toastr.success(message, title, options);
-          break;
+      let config: CdNotificationConfig;
+      if (_.isFunction(arg)) {
+        config = <CdNotificationConfig>arg();
```

Comment:
What about using the newer syntax ``config = arg() as CdNotificationConfig;``? Since Typescript 1.6, the default is as, because <> is ambiguous and unavailable in .tsx files.